### PR TITLE
chore: refactor development setup

### PR DESCRIPTION
**Pull request by afuetterer**
Related issue: #7, #8

## Tasks
- [x] enable CI
- [x] squash commits
- [x] init.py only contains `version = "1.0.1"`
- [x] create atomic commits
- [x] deal with E501 Line too long, break lines or ignore?

### CODE REVIEW

File: rdmo_plugins/__init__.py [hunk 1]
```diff
@@ -1,8 +1 @@
-__title__ = 'rdmo-plugins'
-__version__ = '1.0.0'
-__author__ = 'Jochen Klar'
-__email__ = 'mail@jochenklar.de'
-__license__ = 'Apache-2.0'
-__copyright__ = 'Copyright 2020 RDMO-Arbeitsgemeinschaft'
-
-VERSION = __version__
+VERSION = __version__ = "1.0.0"
```
Comment (afuetterer) [inline rdmo_plugins/__init__.py:1]: I used the same definition style as the rdmo main repo does.
What is `VERSION` actually used for?

File: rdmo_plugins/__init__.py [hunk 2]
```diff
@@ -1,8 +1 @@
-__title__ = 'rdmo-plugins'
-__version__ = '1.0.0'
-__author__ = 'Jochen Klar'
```
Comment (afuetterer) [inline rdmo_plugins/__init__.py:3]: Should I put your name as author and your email in the pyproject.toml? Or `{name = "RDMO Arbeitsgemeinschaft", email = "rdmo-team@listserv.dfn.de"},` like the main repo does?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dynamic = [
+  "version",
+]
+# dependencies = []
```
[the remark this answers is not in the document]
Comment (MyPyDavid) [reply pyproject.toml]: yes, it imports classes and functions directly from `rdmo`

[the remark this answers is not in the document]
Comment (afuetterer) [reply pyproject.toml]: Thanks, I thought so too.

**Comment by afuetterer** [comment]
Are you allowed to enable CI for this repo @MyPyDavid? I can't do it.

**Comment by MyPyDavid** [comment]
I don't see a button with  "Workflow(s) awaiting approval", maybe it needs to "Ready for Review" first?
I can check it

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by afuetterer** [comment]
I guess there needs to be .github/workflows on master?

**Review by MyPyDavid — APPROVED** [review]
approve workflow

**Comment by MyPyDavid** [comment]
> I guess there needs to be .github/workflows on master?

it's there, I dont know what else to try now

**Comment by afuetterer** [comment]
Needed a rebase. Now CI runs, thanks. I will remove your django.yml, okay? There is no manage.py in this repo.

**Comment by MyPyDavid** [comment]
> Needed a rebase. Now CI runs, thanks. I will remove your django.yml, okay? There is no manage.py in this repo.

yes thanks, made a mistake with the `django.yml`, now its running!

File: rdmo_plugins/__init__.py (hunk 2 above)
Comment (MyPyDavid) [reply rdmo_plugins/__init__.py:3]: I would guess so, yes

File: rdmo_plugins/__init__.py (hunk 1 above)
Comment (MyPyDavid) [reply rdmo_plugins/__init__.py:1]: only when it would be released as package on pip I guess,
think you could bump it to 1.0.1

Comment (jochenklar) [reply rdmo_plugins/__init__.py:1]: I think `VERSION` is some ancient convention for `__version__` I added added at one point and never thought about it.

File: rdmo_plugins/__init__.py (hunk 2 above)
Comment (jochenklar) [reply rdmo_plugins/__init__.py:3]: Yes please, only the version should stay in the `__init__.py` file.

File: rdmo_plugins/__init__.py (hunk 1 above)
Comment (afuetterer) [reply rdmo_plugins/__init__.py:1]: I see. If VERSION is not used anywhere, then I propose changing this to only:

```python
__version__ = "1.0.1"
```

Comment (jochenklar) [reply rdmo_plugins/__init__.py:1]: Here its even more complicated: https://github.com/django/django/blob/main/django/__init__.py

**Comment by afuetterer** [comment]
How should this be tested?
I can install rdmo and rdmo-plugins in CI and add a few tests. But I need the testing setup and the fixtures from the main rdmo repo. 

Should I copy and paste?
Should the testing setup and fixtures become their own repo?

**Comment by jochenklar** [comment]
You need the stuff in https://github.com/rdmorganiser/rdmo-catalog to thest this plugin properly. We could clone and setup rdmo in the CI of this repo here, ingest the RDMO catalog (and domain, etc.) and run a few exports/imports.

File: rdmo_plugins/__init__.py (hunk 1 above)
Comment (afuetterer) [reply rdmo_plugins/__init__.py:1]: Nice. You want to do it like that?

**Comment by afuetterer** [comment]
Ruff complains about lines being too long.

**Comment by afuetterer** [comment]
> You need the stuff in https://github.com/rdmorganiser/rdmo-catalog to thest this plugin properly. We could clone and setup rdmo in the CI of this repo here, ingest the RDMO catalog (and domain, etc.) and run a few exports/imports.

I don't understand.

Can you explain what needs to be done?

**Comment by afuetterer** [comment]
I would like to merge this PR. I am going to ignore the long lines warnings.

**Comment by afuetterer** [comment]
Can I merge this @MyPyDavid @jochenklar?

**Comment by jochenklar** [comment]
I have no time to check this right now, but I trust you :D

**Comment by afuetterer** [comment]
I would merge, but I don't have the permissions on this repo.

**Pull request merged**